agent/connect/testing_ca.go

```diff
@@ -175,7 +202,11 @@ func testLeaf(t testing.T, service string, root *structs.CARoot, keyType string,
 	}
 
 	sigAlgo := x509.ECDSAWithSHA256
-	if keyType == "rsa" {
+	rootKeyType, _, err := KeyInfoFromCert(caCert)
```

Comment:
Maybe it would be a good idea to introduce a `SigAlgoForKeyType` helper like `SigAlgoForKey` that does this similar operation rather than scattering these keytype conditionals around.

Reply:
Done in next update